### ISSUE TITLE
Guidance on polylabel tolerance

### ISSUE THREAD

**Issue by sgillies** [Toblerity/Shapely#402]
Follow up on #395.

@allanadair before the 1.6 release we'll need to figure out how to use the tolerance value to get the best label points. See in the image below the difference between using the default (1.0) and a value of 20.0 where the polygon is roughly 20x20 units in size.

![polylabel](https://cloud.githubusercontent.com/assets/33697/17696623/e6f642c6-63af-11e6-8cdc-c57d2d818a1a.png)

**Comment by sgillies** [comment]
Nevermind, I'm thinking incorrectly about this. 1.0 is a fine default.

**Comment by mjmdavis** [comment]
@sgillies in my tests this default does not work very well for geodata.

According to the documentation, the tolerance "represents the highest resolution in units of the input geometry that will be considered for a solution.". This suggests that one needs to ensure that the chosen tolerance is on the order of the dimensions of the features you want the centre of.

I tried to plot what this meant for lat/long: 
![tolerance_distance](https://cloud.githubusercontent.com/assets/1398057/22285929/d52926fe-e2ed-11e6-8e96-57c6b14d01b3.png)

For the geodata I am using, I found that to get accurate results, I needed a value of 1e-5 for tolerance. This translates into a resolution of 1m.

It would be helpful to include some extra commentary in the documentation to help users understand this. Naive tolerance values can produce incorrect results. See the examples here with the different tolerances.

![2017-01-25_11-20-52](https://cloud.githubusercontent.com/assets/1398057/22286718/6c1c8afe-e2f0-11e6-8b3b-d5a48b6631ba.png)

![2017-01-25_11-20-36](https://cloud.githubusercontent.com/assets/1398057/22286738/79d17f24-e2f0-11e6-8c7f-92d628d03087.png)

![2017-01-25_11-19-48](https://cloud.githubusercontent.com/assets/1398057/22286747/7f160c2a-e2f0-11e6-977b-2f8213173b56.png)

Perhaps we could modify the algorithm to provide a guarantee that the returned point is at least within the polygon. Progressively increasing the tolerance until such a time as the result is within the boundary.
